code/game/machinery/vitals_monitor.dm

```diff
@@ -50,76 +72,157 @@
 				breathing = "normal"
 			else if(lungs.breath_fail_ratio < 1)
 				breathing = "shallow"
-		
+
 		to_chat(user, SPAN_NOTICE("Breathing: [breathing]"))
+	if(connected_optable)
+		to_chat(user, SPAN_NOTICE("Connected to adjacent [connected_optable]."))
 
 /obj/machinery/vitals_monitor/Process()
 	if(QDELETED(victim))
-		victim = null
+		update_victim()
 	if(victim && !Adjacent(victim))
-		victim = null
-		update_use_power(POWER_USE_IDLE)
+		update_victim()
+	if(connected_optable && !Adjacent(connected_optable))
+		update_victim()
+		update_optable()
 	if(victim)
 		update_icon()
-	if(beep && victim && victim.pulse())
-		playsound(src, 'sound/machines/quiet_beep.ogg')
-	
+
+/obj/machinery/vitals_monitor/proc/update_victim(var/new_victim = null)
+	var/old_victim = victim
+	victim = new_victim
+	if(victim)
+		visible_message(SPAN_NOTICE("\The [src] is now showing data for \the [victim]."))
```

Comment:
Why does this need a public broadcast message every single time the proc is called? At the very least it should only show if the victim has changed.

Reply:
In practice that's exactly how this works. update_victim() is only called repeatedly as in the QDELETED case outlined above.